### DISCUSSION TITLE
Too many access authorizations when starting mail client

### DISCUSSION THREAD

**Discussion by miquecg** [keepassxreboot/keepassxc#8082] [Q&A]
Hi,

I'm using KeePassXC as my secret provider for storing mail credentials. When starting the mail client I get prompted for every account and calendar I have configured. It looks like all authorization dialogs are opened at once but hidden one behind another, so I end up clicking "Allow" several times.

I was expecting that "Allow everything and future" with remember checkbox would handle this scenario, as all requests are coming from same PID, but I guess that would require to open dialogs one by one so you can just click allow on the first.

I'm aware that these authorizations are only valid while both KeePassXC and application are active, but at least I'd like to click on allow just once.

**Comment by droidmonkey** [comment]
This is an interesting case, and I agree if we already have a prompt for a particular pid we shouldn't show another one until it's answered. We might even be able to just keep adding to the one that is shown. @Aetf what are your thoughts?

It would probably also be useful to be able to permanently allowlist a process/hash, I belive this is already requested though.

**Reply by miquecg** [reply]
Yup. These are the issues I've found so far regarding the "remember me" issue:
- #6458 
- #7571

**Reply by Aetf** [reply]
Blocking prompts for the same pid/client sounds reasonable to me. I created #8084 for this.

I don't think adding new entries to an existing prompt after it is shown is a good idea, though.
It is surprising especially if the new entry is added right before you click allow.

I mentioned in the issue that we could use a small timeout before showing the prompt to collect successive requests.

**Reply by Aetf** [reply]
Regarding client fingerprinting, I'm going to experiment with it in a few weeks and see where I can get. Can't promise anything yet.

**Reply by droidmonkey** [reply]
I like the idea of a small delay to collect successive asks from the same pid
